client.go

```diff
@@ -60,56 +114,69 @@ type UnexpectedResponseError struct {
 // New returns a client.
 // It takes the url of a rspamd instance, and configures the client with Options which are closures.
 func New(url string, options ...Option) *client {
-	client := &client{
-		client: resty.New().SetHostURL(url),
-	}
+	cl := NewFromClient(resty.New().SetHostURL(url))
 
 	for _, option := range options {
-		err := option(client)
+		err := option(cl)
 		if err != nil {
-			log.Fatal("failed to configure client")
+			log.Fatal("failed to configure cl")
```

Comment:
nitpick: this probably should say client so it makes more sense to whoever using this library what failed instead of `cl` which is kinda ambigous?